packages/hardhat-ethers/README.md

```diff
@@ -42,7 +42,16 @@ automatically connected to the selected network.
 These helpers are added to the `ethers` object:
 
 ```typescript
-function getContractFactory(name: string, signer?: ethers.Signer): Promise<ethers.ContractFactory>;
+interface LibraryLinks {
+  [libraryName: string]: string;
+}
+
+interface FactoryOptions {
+  signer?: ethers.Signer;
+  libraryLinks?: LibraryLinks;
+}
+
+function getContractFactory(name: string, signerOrFactoryOptions?: ethers.Signer | FactoryOptions): Promise<ethers.ContractFactory>;
```

Comment:
Let's split this into three different signatures here, as this is documentation.

```ts
function getContractFactory(name: string): Promise<ethers.ContractFactory>;

function getContractFactory(name: string, signer: ethers.Signer): Promise<ethers.ContractFactory>;

function getContractFactory(name: string, factoryOptions?: FactoryOptions): Promise<ethers.ContractFactory>;
```